### PR TITLE
DCP 2 Migration

### PR DESCRIPTION
**Pull request by wesbos**

### CODE REVIEW

**Comment by paulrobinson** [comment]
I'm getting zero results for: http://it-developers-pr.stage.redhat.com/pr/550/build/1149/resources/#/sys_type=quickstart&query=

**Comment by paulrobinson** [comment]
Also zero results for: http://it-developers-pr.stage.redhat.com/pr/550/build/1149/resources/#/sys_type=jbossdeveloper_sandbox&query=&from=0

**Comment by paulrobinson** [comment]
Resources should be ordered reverse-chronilogical on http://it-developers-pr.stage.redhat.com/pr/550/build/1149/resources/. I.e such that the newest ones appear first.

**Comment by paulrobinson** [comment]
There are no events showing on: http://it-developers-pr.stage.redhat.com/pr/550/build/1149/events/

**Comment by lukas-vlcek** [comment]
Couple of comments:

- The initial query seems to be exectured several time on page load. For example the [Resources](/resources) page does it four times. Three XHR calls are cancelled and the last one is finished.
- There are still some calls to Searchisko 1, i.e. on staging <http://dcpbeta-searchisko.rhcloud.com>. I recommend to replace Searchisko 1 URLs with Searchisko 2 URLs and checking if there are any issues.
- When the `publish_date` value is changed in the URL this change is not propagated/applied the same way it works for `query` and `sys_type` URL parameters. Sometime this can lead to confusing situations.

screenshot 2015-11-10 11 01 02

The above screenshot shows cancelled requests and query to old Searchisko 1 (the last request in the list).

**Comment by wesbos** [comment]
@lukas-vlcek A coulple questions about the API that I could not find in the docs

* How do I sort descending? IE newest developer materials first?
* There are no quickstarts, no events and no experimental - can we get all the data into the DCP2 instance so we can properly test it?

**Comment by lukas-vlcek** [comment]
@wesbos Sorting by date is not possible yet, however, it should be easy to add (let me do it). And I will check if there are running indexers on your testing DCP2 instance.

**Comment by lukas-vlcek** [comment]
@wesbos see progress of https://github.com/searchisko/configuration/issues/34 for date sorting feature.

File: Gemfile.lock [hunk 1]
```diff
@@ -1,6 +1,5 @@
-GIT
-  remote: git://github.com/awestruct/aweplug.git
-  revision: cdd64b67aee9c0bcd7f4f257ce42543dd4c5a7b1
+PATH
+  remote: ~/Dropbox/redhat/aweplug/
```
Comment (paulrobinson) [inline Gemfile.lock]: Needs an Aweplug PR for this

**Comment by paulrobinson** [comment]
Needs a rebase

**Comment by wesbos** [comment]
@paulrobinson This should be ready for a code review. A few things to note:

1. There doesn't seem to be any events in DCP2  - is this right? @lukas-vlcek 
2. @lukas-vlcek Can you double check I have the URLs correct in `site.yml` ?

**Comment by paulrobinson** [comment]
Looks like the community page is broken. Our new tests spotted it :-)

To see the results: 

1. visit: https://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/developers.redhat.com-pull-player-executor/1219/
2. Click on the build number next to the linked 'frontend-acceptance-tests' job
3. Look at the console output
4. Notice that the Cucumber test that checks for results on /community is failing.
5. Visit the page, and see why: http://it-developers-pr.stage.redhat.com/pr/550/build/1219/projects/

**Comment by lukas-vlcek** [comment]
@wesbos to me URLs to Searchisko 2 look correct in the file `_config/site.yml`. No Searchisko 1 leftovers.

**Comment by lukas-vlcek** [comment]
@paulrobinson I was checking site at <http://it-developers-pr.stage.redhat.com/pr/550/build/1219/projects/> and it seem to me that the site is making XHR request to <http://10.3.9.71:32778/> server which is old Searchisko (not Searchisko 2). Am I correct?

The full URL is it calling is <http://10.3.9.71:32778/v2/rest/search/suggest_project_name_ngram_more_fields?sort=sys_title&query=> which indicates it assumes Searchisko 2.

**Comment by wesbos** [comment]
@lukas-vlcek what is the URL for the new searchisko in staging? The community pages works locally with the DCP2 url or development

**Comment by lukas-vlcek** [comment]
@wesbos http://dcp.stage.jboss.org/ but I think it is behind VPN.

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
I think the error is because you need to change: https://github.com/redhat-developer/developers.redhat.com/blob/167120adc2ce305d7cd0eda3c0cd57812506a34a/_ext/events.rb#L47

'jbossdeveloper_event' is now 'event'.

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
@wesbos please rebase this with master cc @paulrobinson

**Comment by ollyjshaw** [comment]
retest this please

**Comment by wesbos** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by wesbos** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
Outstanding stuff:

* Enforce list view on /resources
* Slider grip is centred on /resources when 'all' is selected
* Sliding the grip to the left causes 'It appears we're unable to access this data right now. Look at @jbossorg to see if there is scheduled maintenance, or try again shortly.' to show
* Selecting 'quickstart' in /resources returns nothing
* Selecting 'early access' in /resources returns nothing
* /events shows nothing. Should be 1. The AJAX call seems to be correct, so probably a rendering issue.
* Connectors request is not firing on http://it-developers-pr.stage.redhat.com/pr/550/build/190/products/fuse/connectors/

Please also search for all the places that DCP is called and check they work.

**Comment by wesbos** [comment]
Okay - a few things:

1) Quickstarts and Early Access are not returning anything - nothing wrong with the URL we are hitting. I've sent @lukas-vlcek an email about this.
2) There are no bugs in the events page, it's just that there is only a single event in the DCP, and that one event does not have a start date assigned to it, so it doesn't show up.

**Comment by wesbos** [comment]
Update: @lukas-vlcek is looking into the quickstarts and early access not returning any data

**Comment by wesbos** [comment]
just rebased this - should be good to merge after the tests.

**Comment by paulrobinson** [comment]
Unfortunately, we can't merge until the Docker instance of the DCP is upgraded.

**Comment by ollyjshaw** [comment]
needs a rebase

**Comment by paulrobinson** [comment]
Closing. I'll re-open with the Docker changes.

**Pull request closed without merging**